tracking/tracking_ocp.py

```diff
@@ -0,0 +1,178 @@
+
+        self.data = C3dData(c3d_path, self.biorbd_model)
+
+        self.final_time = self.data.get_final_time() if final_time is None else final_time
+        self.n_shooting_points = n_shooting_points
+        self.nb_iteration = nb_iteration
+        self.data_loaded = LoadData(self.biorbd_model, c3d_path, self.q_file, self.qdot_file)
+
+        self.q_ref, self.qdot_ref, self.markers_ref = self.data_loaded.get_experimental_data(
+                                                                                [self.n_shooting_points],
+                                                                                [self.final_time],
+                                                                                with_floating_base=with_floating_base)
+        self.ocp = None
+        self.prepare_ocp()
+
+    def prepare_ocp(self):
+        nb_q = self.biorbd_model.nbQ()
+        nb_qdot = self.biorbd_model.nbQdot()
+
+        # Add objective functions
+        objective_functions = ObjectiveList()
+        objective_functions.add(ObjectiveFcn.Lagrange.MINIMIZE_CONTROL, key="tau", weight=10)
+        objective_functions.add(ObjectiveFcn.Mayer.TRACK_MARKERS, weight=100000, target=self.markers_ref[0], node=Node.ALL)
+        if self.with_floating_base:
+            objective_functions.add(ObjectiveFcn.Lagrange.MINIMIZE_STATE, weight=10, key="qdot",
+                                    index=[0, 1, 2, 3, 4, 5])
+        else :
+            objective_functions.add(ObjectiveFcn.Lagrange.MINIMIZE_STATE, weight=10, key="qdot")
+
+        # Dynamics
+        dynamics = DynamicsList()
+        dynamics.add(DynamicsFcn.TORQUE_DRIVEN)
+
+        # Path constraint
+        x_bounds = BoundsList()
+        x_bounds.add(bounds=QAndQDotBounds(self.biorbd_model))
+
+        # Initial guess
+        init_x = np.zeros((nb_q + nb_qdot, self.n_shooting_points + 1))
+        init_x[:nb_q, :] = self.q_ref[0]
+        init_x[nb_q: nb_q + nb_qdot, :] = self.qdot_ref[0]
+
+        x_init = InitialGuessList()
+        x_init.add(init_x, interpolation=InterpolationType.EACH_FRAME)
+
+        # Define control path constraint
+        u_bounds = BoundsList()
+        u_init = InitialGuessList()
+        tau_min, tau_max, tau_init = -100, 100, 0
+        u_bounds.add(
+            [tau_min] * self.biorbd_model.nbGeneralizedTorque(),
+            [tau_max] * self.biorbd_model.nbGeneralizedTorque(),
+        )
+        u_init.add([tau_init] * self.biorbd_model.nbGeneralizedTorque())
+
+        # ------------- #
+        self.ocp = OptimalControlProgram(
+            biorbd_model=self.biorbd_model,
+            dynamics=dynamics,
+            n_shooting=self.n_shooting_points,
+            phase_time=self.final_time,
+            x_init=x_init,
+            u_init=u_init,
+            x_bounds=x_bounds,
+            u_bounds=u_bounds,
+            objective_functions=objective_functions,
+            ode_solver=self.ode_solver,
+            n_threads=6
```

Comment:
Please put n_thread has an argument and an attribute of the object, by default 6,

Reply:
done